docs/posts/introduction-to-ballerina-0.970.md

```diff
@@ -0,0 +1,220 @@
+
+### Fork/Join
+The fork/join construct in Ballerina is used in order to split (fork) the current function execution to multiple workers, do some processing in parallel, and join together the results of the workers to a single execution again.
+
+### Lock
+Ballerina locks are used for concurrency management, encapsulating a block of statements will acquire the locks for each global or service level variable reference used within those block of statements.
+
+
+## Functions
+Ballerina functions operate the same way as any other language. It is a mechanism to create a reusable unit of functionality within a program.  
+A function pointer is a Ballerina type that allows you to use functions as variables, arguments to functions, and function return values. 
+Lambdas are a syntactic shortcut for defining inline functions. In comparison to a normal function definition, the only missing part is the name. 
+
+
+# Syntax
+
+Ballerina’s underlying language semantics were designed by modeling how independent parties communicate via structured interactions. Subsequently, every Ballerina program can be displayed as a sequence diagram of its flow with endpoints, including synchronous and asynchronous calls. Ballerina’s syntax has both textual and graphical representation designed around sequence diagrams,therefore, the way a developer thinks when writing Ballerina code encourages strong interaction best practices.
+
+## Textual Syntax
+Ballerina’s textual syntax is largely inspired by C, Java, and Go languages. The key language constructs in Ballerina are Function, Worker and Service. 
+
+## Graphical Syntax
+Ballerina’s graphical syntax resembles a sequence diagram. The control flow within a worker is represented with flow diagram based elements. Graphical representation of a service is represented with network interactions.
+
+Ballerina platform comes with the Composer IDE, which allows you to edit and view Ballerina programs graphically and textually. VS Code plugin can be also used to view Ballerina programs graphically.
+
+# Integration Specialization 
+Ballerina has first class support for services and endpoints. HTTP/HTTP2, WebSockets, WebSub, gRPC, and JMS are some of the available service types. These services are exposed via listener endpoints, which can be secured and monitored. Client endpoints connect to different types of external endpoints and they are inherently resilient. Additionally, commonly used integration message formats, such as XML and JSON, are built-in to the type system of the language. 
+
+In the context of integration specialization, the following are the released features.
+
+## HTTP 1.x/2 
+Ballerina includes HTTP 1 and 2 support with server and client endpoints. 
+
+## WebSockets
+WebSocket client/server endpoints. 
+
+## Resiliency
+Support for network resiliency with Circuit breaker, retry, timeout, load balancing and failover. 
```

Comment:
I feel 'circuit breaker' can be lowercase here. 

Add a serial comma before 'and'. We do this normally for lists.